EIPS/eip-FOCIL.md

```diff
@@ -26,10 +26,85 @@ FOCIL is a simple committee-based design improving upon previous IL mechanisms o
+    entries: List[InclusionListEntry, MAX_TRANSACTIONS_PER_INCLUSION_LIST]
+```
+
+```
+class InclusionListAggregateEntry(Container):
+    from_address: ExecutionAddress
+    gas_limit: uint64
+    bitlist: List[byte, IL_COMMITTEE_SIZE]
+```
+
+```
+class InclusionListAggregate(Container):
+    slot: Slot
+    proposer_index: ValidatorIndex
+    message: List[InclusionListAggregateEntry, MAX_TRANSACTIONS_PER_INCLUSION_LIST_AGGREGATE]
+```
+
 ### Consensus layer
 
 #### High-level overview
 
+**Beacon chain state transition spec:**
+
+- ***New** `inclusion_list` object:* Introduce a new `inclusion_list` for the IL committee to submit and nodes to process.
+- ***Modified** `BeaconBlockBody`:* Modified to include the inclusion list aggregate for that slot.
+- ***Modified** `process_execution_payload` function:* Update this process to include checks for the inclusion list evaluation.
+
+**Beacon chain P2P spec:**
+
+- ***New** gossipnet and validation rules for inclusion list:* Define new rules for handling the inclusion list in the gossip network and validation.
+- ***New** RPC request and response network for inclusion list:* Establish a new network for sending and receiving inclusion lists.
+- ***New** gossipnet and validation rules for inclusion list aggregate:* Define new rules for handling the inclusion list aggregate in the gossip network and validation.
+- ***New** RPC request and response network for inclusion list aggregate:* Establish a new network for sending and receiving inclusion lists aggregate.
+
+**Validator spec:**
+
+- ***New** duty for preparing `inclusion_list`:* Inclusion list committee members to prepate and sign their respective local inclusion list.
+- ***New** proposer duty for aggregating `inclusion_list`:* Proposer to prepare an inclusion list aggregate and sign it.
+- ***Modified** duty for `BeaconBlockBody`:* Update the duty to prepare the beacon block body containing `inclusion_list_aggregate` and satisfying transaction entries if block is not full.
+
+### Execution layer
+
+- ***New** `get_inclusion_list`:* Introduce a new function for the IL committee to retreive inclusion lists.
+- ***Modified** `forkchoice_updated`:* Update the function with a `payload_attribute` to include the inclusion list aggregate as part of the attribute.
+- ***Modified** `new_payload`:* Update the function for EL clients to verify that `payload_transactions` satisfy `payload.inclusion_list_aggregate`.
+- ***New** validation rules:* Implement new validation rules based on the changes introduced in the Execution-API spec.
+
+### Fork choice changes
+
+Validators performs additional checks (on top of existing validations) before attesting on a block. 
+
+- If the aggregate inclusion list is not made available before the deadline, the block is valid.
+- If the aggregate inclusion list is made available before the deadline:
+    - If the payload doesn't include those transactions, the block is considered as invalid.
```

Comment:
those transactions => transactions in the inclusion list aggregate

remove "as" before invalid

Reply:
I think we should only treat the word "invalid" as the block fails state transition vs. in this case the block would just be non canonical because no honest attester will vote for it